src/Sylius/Component/Promotion/Generator/CouponGenerator.php

```diff
@@ -68,13 +72,19 @@ public function generate(PromotionInterface $promotion, Instruction $instruction
     /**
      * {@inheritdoc}
      */
-    public function generateUniqueCode()
+    public function generateUniqueCode($codeLength)
     {
         $code = null;
 
+        if (1 > $codeLength || 40 < $codeLength) {
```

Comment:
Moreover when i implement this in model i will get conflicts with symfony form component cuz i will get exception before validate method will be called but I may be wrong.